### ISSUE TITLE
Curious about the benefit to use online distillation for sam predict? 

### ISSUE THREAD

**Issue by YZsZY** [ashawkey/Segment-Anything-NeRF#1]
Hello author! I would like to ask for your thoughts when implementing the code, why gt_samvit use rendered image for prediction, instead of using gt_image directly as input to sam?
https://github.com/ashawkey/Segment-Anything-NeRF/blob/07bdc5d0e018aad71849fb7be6160b23859b5942/nerf/utils.py#L532-L548

I noticed that when generating gt_sam for supervised network learning semantics, it is using the rendered image as input to sam_predictor, why not just use the GT_image input to sam to get the gt_label? Is there any benefit to this self online distillation method?

**Comment by ashawkey** [comment]
@YZsZY Hi, that's because we augment the training cameras so we don't have ground truth images for these novel views. We found only using the training dataset is not enough for learning robust features.

**Comment by YZsZY** [comment]
@ashawkey Thanks for your reply! So generating semantic labels for novel views to expand the sample so that it can help semantic network learning?

**Comment by ashawkey** [comment]
Yes, we found this way makes the SAM features better learned.
